### ISSUE TITLE
#1362 Guest UI: error: "undefined method `id' for nil:NilClass" 

### ISSUE THREAD

**Issue by NishantUpadhyay-BTC** [NishantUpadhyay-BTC/BLISS-Issue-Tracking#292]
Just got this error in red at the top of the screen while testing the GP ticket #1318: "undefined method `id' for nil:NilClass"

What I did was:

While logged in as Gui Guest024:
Select dates for a period where we do have availability.
Select one guest and choose "GP"
Change # of guests to 2.
Check for available lodgings.
Clicked "Reserve Now" for a non-plumbing cabin (I think).
Verified that there was no GP.
Clicked "Cancel and return to Availability"

Chose a new date range (I think it was June 28th for 1 night)
Chose 1 person.
Did not assign gender.
Clicked on "Check for Available Lodgings"

**Comment by NishantUpadhyay-BTC** [comment]
I'm getting this error also when I've completed a reservation, choose to make another, choose dates and then click "reserve now" on an available lodging.

**Comment by NishantUpadhyay-BTC** [comment]
I was making several reservations today using the "Gui Guest024" ID. After a while I was getting the above mentioned error many times. I also noticed I was getting the option to "Continue to Existing" reservation, even though I had completed my open reservation.

I went back into the Office UI and checked for reservations in progress. I found three open reservations in progress for Gui Guest 024, so I deleted those. I was still getting the "existing reservation" option when trying to reserve lodgings for a new reservation. So I logged out of Gui Guest024 account, confirmed that there were no reservations in progress and no unconfirmed reservations for Guest024 and then proceeded again to check for available lodgings and make a new reservation. I chose "Reserve Now" and logged in as Guest024, then was immediately returned to the PR Availability Page with the same error listed above. Checking back to the Office UI, I found Guest024 now had a reservation in progress- but any attempt to return to that reservation from the Guest UI resulted in returning to the PR availability page with that same error. If I chose to "create new" instead, then I STILL went back to the PR Availability page and got the same error - and now yet another new reservation was created for Guest024!

Then I tried creating a new user from guest record for "Gui Guest098" and logged in as that user. Upon login, I was once again returned to PR availability page with "undefined method" error at the top.

So, we have a few things going on here:

First - something is creating the "undefined method" error. I still don't know where that is coming from. Note that I also tried clearing my browser cache (Chromium) and that didn't appear to correct anything.

Second: When user is given the option to go to an existing reservation or creating a new one, and chooses to go to existing reservation, it should always go to the Reservation Details page for that reservation. There is no reason why clicking this button should ever take us back to the Personal Retreat Availability page - unless the Reservation timer has run out, in which case we should be getting the timeout message... which we're not.

Third: When user is given the option to return to an exiting reservation or create a new one and chooses "Create a New Reservation" that ought to Cancel and delete the existing reservation in progress, create a new one, and then take us to the Guest Details page for that reservation.
